### DISCUSSION TITLE
Liquibase initialization fails on Percona XtraDB Cluster

### DISCUSSION THREAD

**Discussion by x-c4mp3r** [keycloak/keycloak#26895] [Q&A]
### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

dist/quarkus

### Describe the bug

Percona-XtraDB-Cluster need a primary key in DATABASECHANGELOG table.
`Caused by: liquibase.exception.LiquibaseException: liquibase.exception.DatabaseException: Percona-XtraDB-Cluster prohibits use of DML command on a table (keycloak.DATABASECHANGELOG) without an explicit primary key with pxc_strict_mode = ENFORCING or MASTER [Failed SQL: (1105) INSERT INTO keycloak.DATABASECHANGELOG (ID, AUTHOR, FILENAME, DATEEXECUTED, ORDEREXECUTED, MD5SUM, `DESCRIPTION`, COMMENTS, EXECTYPE, CONTEXTS, LABELS, LIQUIBASE, DEPLOYMENT_ID) VALUES ('1.0.0.Final-KEYCLOAK-5461', 'sthorger@redhat.com', 'META-INF/jpa-changelog-1.0.0.Final.xml', NOW(), 1, '8:bda77d94bf90182a1e30c24f1c155ec7', 'createTable tableName=APPLICATION_DEFAULT_ROLES; createTable tableName=CLIENT; createTable tableName=CLIENT_SESSION; createTable tableName=CLIENT_SESSION_ROLE; createTable tableName=COMPOSITE_ROLE; createTable tableName=CREDENTIAL; createTable tab...', '', 'EXECUTED', NULL, NULL, '4.8.0', '5587500563')`

### Version

20.0.0

### Expected behavior

database initialization completes without any error if primary key is needed for the database

### Actual behavior

When I follow the liquibase documentation and creating a table manually the next problem is:
jpa-changelog-1.0.0.Final.xml:
```
<changeSet author="sthorger@redhat.com" id="1.0.0.Final-KEYCLOAK-5461">
```
jpa-changelog-1.0.0.Final-db2.xml:
```
<changeSet author="sthorger@redhat.com" id="1.0.0.Final-KEYCLOAK-5461">
```
the id field is the same, so initializing is getting error here.

Same problem for the following files:

```
db2-jpa-changelog-1.2.0.Beta1.xml
db2-jpa-changelog-1.2.0.CR1.xml
db2-jpa-changelog-1.4.0.xml
db2-jpa-changelog-1.8.0.xml -> 1.8.0 and 1.8.0-2
db2-jpa-changelog-1.9.1.xml
jpa-changelog-12.0.0.xml -> map-remove-ri conflicts with jpa-changelog-11.0.0.xml
```

### How to Reproduce?

Run a keycloak 20.0.0 (or newer) docker image which connects to a version 8 Percona-XtraDB-Cluster. The database initialization will fail.

**Comment by ahus1** [comment]
Thank you for reporting this. The database Percona-XtraDB-Cluster is not supported, so problems like the one you describe can occur. See https://www.keycloak.org/server/db for the tested databases.

The Liquibase docs recommend to create a primary key for the table manually, see https://docs.liquibase.com/concepts/tracking-tables/databasechangelog-table.html

From your description above, you mention creating a table, but not an index. Please double check if you can proceed further in the installation after adding the index.

**Comment by x-c4mp3r** [comment]
Dear @ahus1.

I looked and used this manual: https://contribute.liquibase.com/extensions-integrations/directory/database-tutorials/percona/percona-xtradb-cluster/
I created the DATABASECHANGELOGLOCK and DATABASECHANGELOG tables with PRIMARY KEY (ID).
If I use this the mentioned xml files have conflicting id fields.

> Please double check if you can proceed further in the installation after adding the index

Sorry that I'm misunderstood you but what index do you mean ?
